### PR TITLE
feat: :arrow_up: migrate to Angular 16

### PR DESCRIPTION
**Pull request by Nicoss54**
This PR make the library compatible with Angular 16

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,30 +11,30 @@
...
     "@types/jasmine": "~4.0.0",
     "jasmine-core": "~4.1.0",
     "karma": "~6.3.0",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage": "~2.2.0",
     "karma-jasmine": "~5.0.0",
     "karma-jasmine-html-reporter": "~1.7.0",
-    "ng-packagr": "^15.1.1",
+    "ng-packagr": "^16.0.0",
     "typescript": "~4.9.5"
```
Comment (jornare) [inline package.json:38]: Should it not have TypeScript v5?

Comment (Nicoss54) [reply package.json:38]: the minimal typescript version for Angular 16 is 4.9, thats why the update schematics don't change it.
But we can upgrade to v5

**Review by mgechev — APPROVED** [review]

**Comment by mgechev** [comment]
Thank you for the PR!

**Comment by mgechev** [comment]
Released as v0.4.2.

**Pull request merged**